src/pkg/services/m365/api/channels.go

```diff
@@ -1 +1,376 @@
+		return nil, clues.New("channel not found").WithClues(ctx)
+	}
+
+	// We only allow the api to match one calendar with the provided name.
+	// If we match multiples, we'll eagerly return the first one.
+	logger.Ctx(ctx).Debugw("calendars matched the name search", "calendar_count", len(gv))
+
+	// Sanity check ID and name
+	cal := gv[0]
+	container := ChannelsDisplayable{Channelable: cal}
+
+	if err := graph.CheckIDAndName(container); err != nil {
+		return nil, clues.Stack(err).WithClues(ctx)
+	}
+
+	return container, nil
+}
+
+func (c Channels) PatchChannel(
+	ctx context.Context,
+	teamID, containerID string,
+	body models.Channelable,
+) error {
+	_, err := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Patch(ctx, body, nil)
+
+	if err != nil {
+		return graph.Wrap(ctx, err, "patching event calendar")
+	}
+
+	return nil
+}
+
+// ---------------------------------------------------------------------------
+// message
+// ---------------------------------------------------------------------------
+
+// GetItem retrieves a Messageable item.
+func (c Channels) GetMessage(
+	ctx context.Context,
+	teamID, channelID, itemID string,
+	immutableIDs bool,
+	errs *fault.Bus,
+) (serialization.Parsable, *details.GroupsInfo, error) {
+	var (
+		size int64
+	)
+
+	// is preferImmutableIDs headers required here
+	message, err := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(channelID).
+		Messages().
+		ByChatMessageId(itemID).
+		Get(ctx, nil)
+	if err != nil {
+		return nil, nil, graph.Stack(ctx, err)
+	}
+
+	return message, MessageInfo(message, size), nil
+}
+
+func (c Channels) PostMessage(
+	ctx context.Context,
+	teamID, containerID string,
+	body models.ChatMessageable,
+) (models.ChatMessageable, error) {
+	itm, err := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Messages().
+		Post(ctx, body, nil)
+	if err != nil {
+		return nil, graph.Wrap(ctx, err, "creating mail message")
+	}
+
+	if itm == nil {
+		return nil, clues.New("nil response mail message creation").WithClues(ctx)
+	}
+
+	return itm, nil
+}
+
+func (c Channels) DeleteMessage(
+	ctx context.Context,
+	teamID, itemID, containerID string,
+) error {
+	// deletes require unique http clients
+	// https://github.com/alcionai/corso/issues/2707
+	srv, err := NewService(c.Credentials)
+	if err != nil {
+		return graph.Stack(ctx, err)
+	}
+
+	err = srv.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Messages().
+		ByChatMessageId(itemID).
+		Delete(ctx, nil)
+	if err != nil {
+		return graph.Wrap(ctx, err, "deleting mail message")
+	}
+
+	return nil
+}
+
+// ---------------------------------------------------------------------------
+// replies
+// ---------------------------------------------------------------------------
+
+// GetReplies retrieves a Messageable item.
+func (c Channels) GetReplies(
```

Comment:
Will this need a pager implementation?

Reply:
We can add this implementation if required in the PR with pager.